qt/traffic_panel.cpp

```diff
@@ -0,0 +1,77 @@
+#include "qt/traffic_panel.hpp"
+#include "qt/traffic_mode.hpp"
+
+#include "std/string.hpp"
+
+#include <QAbstractTableModel>
+#include <QBoxLayout>
+#include <QComboBox>
+#include <QHeaderView>
+#include <QStyledItemDelegate>
+#include <QTableView>
+
+// ComboBoxDelegate --------------------------------------------------------------------------------
+ComboBoxDelegate::ComboBoxDelegate(QObject * parent)
+  : QStyledItemDelegate(parent)
+{
+}
+
+QWidget * ComboBoxDelegate::createEditor(QWidget * parent, QStyleOptionViewItem const & option,
+                                         QModelIndex const & index) const
+{
+  auto * editor = new QComboBox(parent);
```

Comment:
Undone.) 
QComboBox(parent) attaches a new object to a parent and gc can reach it since then.